### DISCUSSION TITLE
Find all hex starting with

### DISCUSSION THREAD

**Discussion by akshaybabloo** [WerWolv/ImHex#1996] [Get Help] [answered]
I have a binary file in which all `55 AA` is start of a row and ends with `55 AA`, which is also a start of the next row. How should I find all data that is starting with `55 AA` and ends with `55 AA`?

**Comment by WerWolv** [comment]
You can use the Find view for this. Use the Binary Pattern option there and enter `55 AA` into its search field. That will show you all occurences

**Comment by paxcut** [comment, marked as answer]
You are missing some details as to how you want to find these sequences and what happens if the magic numbers 0x55,0xAA are not in the data. Assuming that data starts and end with said sequence and assuming you want to write a pattern to find these sequences the following should work:
```cpp
import type.base;
import std.mem;

struct ReadRow {
    $ += 2;
    type::Hex<u8> data[while(($[$]!=0x55 || $[$+1] != 0xAA) && !std::mem::eof())];
};

str seq="\x55\xAA";

std::mem::MagicSearch<seq,ReadRow> find@0;
```
I tested it on some made-up data and it seems to work.
